src/cl_main.cpp

```diff
@@ -6662,6 +6662,124 @@ void ServerCommands::SetLineTextureByID::Execute()
 		CLIENT_PrintWarning( "SetLineTextureByID: unknown texture: %s\n", textureName.GetChars() );
 }
 
+//*****************************************************************************
+//
+void ServerCommands::SetLineTextureOffset::Execute()
+{
+	if ( line->sidedef[side] == NULL )
+		return;
+
+	line->sidedef[side]->textures[position].xoffset = XOffset;
+	line->sidedef[side]->textures[position].yoffset = YOffset;
+}
+
+//*****************************************************************************
+//
+void ServerCommands::SetLineTextureOffsetByID::Execute()
+{
+	int linenum = -1;
+	const fixed_t NO_CHANGE = 32767 << FRACBITS;
+
+	while (( linenum = P_FindLineFromID( lineID, linenum )) >= 0 )
+	{
+		side_t *sidedef = lines[linenum].sidedef[side];
+		if ( sidedef != NULL )
+		{
+			if (( flags & 8 ) == 0 )
+			{
+				if ( XOffset != NO_CHANGE )
+				{
+					if ( flags & 1 ) sidedef->SetTextureXOffset( side_t::top, XOffset );
+					if ( flags & 2 ) sidedef->SetTextureXOffset( side_t::mid, XOffset );
+					if ( flags & 4 ) sidedef->SetTextureXOffset( side_t::bottom, XOffset );
+				}
+
+				if ( YOffset != NO_CHANGE )
+				{
+					if ( flags & 1 ) sidedef->SetTextureYOffset( side_t::top, YOffset );
+					if ( flags & 2 ) sidedef->SetTextureYOffset( side_t::mid, YOffset );
+					if ( flags & 4 ) sidedef->SetTextureYOffset( side_t::bottom, YOffset );
+				}
+			}
+			else
+			{
+				if ( XOffset != NO_CHANGE )
+				{
+					if ( flags & 1 ) sidedef->AddTextureXOffset( side_t::top, XOffset );
+					if ( flags & 2 ) sidedef->AddTextureXOffset( side_t::mid, XOffset );
+					if ( flags & 4 ) sidedef->AddTextureXOffset( side_t::bottom, XOffset );
+				}
+
+				if ( YOffset != NO_CHANGE )
+				{
+					if ( flags & 1 ) sidedef->AddTextureYOffset( side_t::top, YOffset );
+					if ( flags & 2 ) sidedef->AddTextureYOffset( side_t::mid, YOffset );
+					if ( flags & 4 ) sidedef->AddTextureYOffset( side_t::bottom, YOffset );
+				}
+			}
+		}
+	}
+}
+
+//*****************************************************************************
+//
+void ServerCommands::SetLineTextureScale::Execute()
+{
+	if ( line->sidedef[side] == NULL )
+		return;
+
+	line->sidedef[side]->textures[position].xscale = XScale;
+	line->sidedef[side]->textures[position].yscale = YScale;
+}
+
+//*****************************************************************************
+//
+void ServerCommands::SetLineTextureScaleByID::Execute()
+{
+	int linenum = -1;
+	const fixed_t NO_CHANGE = 32767 << FRACBITS;
```

Comment:
This is reimplementing `LS_Line_SetTextureScale`, right? Then, you can should be able to handle it like `LS_Line_SetTextureOffset`, see my comment above.

Reply:
Sounds good, thanks for the suggestion.